wgpu-hal/src/gles/adapter.rs

```diff
@@ -806,6 +806,19 @@ impl crate::Adapter<super::Api> for super::Adapter {
             formats.push(wgt::TextureFormat::Rgba16Float)
         }
         if surface.presentable {
+            let mut formats = vec![
```

Comment:
This shadowing the outer format vec seems very strange.

Reply:
oops, that must have happend when rebasing because of https://github.com/gfx-rs/wgpu/pull/3045
Fixed and disabled the new f16 path - I think we'd need a special path for that since right now it would get the srgb treatment and blit wouldn't work. cc: @jinleili

Reply:
Could you add that path in this PR? Hopefully it shouldn't be that difficult - the #[cfg] on the f16 path feels kinda dirty

Reply:
you were right, the blit actually supports it as is. Went through the motion to test it and renders just fine :)

Reply:
f16 works as is

Reply:
Great!